pageserver/src/aux_file.rs

```diff
@@ -140,6 +143,55 @@ pub fn encode_file_value(files: &[(&str, &[u8])]) -> anyhow::Result<Vec<u8>> {
     Ok(encoded)
 }
 
+/// An estimation of the size of aux files.
+pub struct AuxFileSizeEstimator {
+    aux_file_size_gauge: IntGauge,
+    size: Arc<std::sync::Mutex<Option<isize>>>,
```

Comment:
can this be an atomic?

Reply:
We need something like `Atomic<Option<isize>>`, which is only possible with manually-implemented compare-with-swap. As the patch is not on the critical path, I prefer keep it simple.

Reply:
Good point, it's fine.